File: 0/0/9/009.py
# Copyright (C) 2013 Jorge Aparicio

P = 1000

print(next((P - b - c) * b * c
           for c in range(P // 3 + 1, P // 2)
           for b in range((P - c) // 2 + 1, c)
           if (P - b - c) * (P - b - c) + b * b == c * c))

File: 0/0/9/009.rs
// Copyright (C) 2014 Jorge Aparicio

static P: int = 1000;

fn main() {
  for c in range(P / 3 + 1, P / 2) {
    for b in range((P - c) / 2 + 1, c) {
      let a = P - b - c;

      if a*a + b*b == c*c {
        println!("{}", a * b * c);

        return;
      }
    }
  }
}

